feat: add vault list command

realnits opened the pull request:
Add new command to discover and list vaults in a directory. Supports recursive search and detailed metadata display..

- Add list subcommand with ls/find/search aliases
- Show vault metadata without unlocking
- Support recursive directory scanning
- Display name, namespace, secret count, accessor count
- Add comprehensive documentation

This is one of my first open-source contributions after learning Go for my project.

realnits committed:
feat: add vault list command

Add new command to discover and list vaults in a directory.
Supports recursive search and detailed metadata display.

- Add list subcommand with ls/find/search aliases
- Show vault metadata without unlocking
- Support recursive directory scanning
- Display name, namespace, secret count, accessor count
- Add comprehensive documentation

realnits commented:
Hi @dhanushmuralidharan How about rather than displaying it as detailed text.. we can set it up like a metric / stat value.

In that way, we will have enough space to display the table in efficient way and good readability as well

<img width="671" height="145" alt="Screenshot 2025-12-10 at 11 40 36 AM" src="https://github.com/user-attachments/assets/6874d012-f96e-4ea7-80d2-6e0cd860d7a6" />

sigsegvd commented:
@realnits Sorry but I'm not catching up. Can you please clarify what you mean by _metric/stat value_ ? Also, the table would be just fine if we can wrap the content inside the cell itself as seen in your screenshot.

realnits committed:
Improve vault list command with verbose output and cell wrapping

- Add verbose default list view with detailed vault information
- Show vault name, namespace, secrets count, and accessor information
- Add numbered list with color-coded output for better readability
- Enhance detailed table view with Access List column
- Implement proper cell wrapping to prevent overflow on narrow terminals
- Add summary footer with total statistics
- Fix table readability issues on Windows with larger fonts
- Update documentation to reflect new output formats

realnits commented:
@sigsegvd Hi Dhanush, I've wrapped up the table and updated it and here is the attachedimage on how does it look now

```sh
sniith@ININHGSPRIC95BF slv % ./slv vault list -r -l
Found 1 vault(s):

┌─────┬────────────────────────────────┬────────────┬────────────┬──────────┬────────────┬─────────────────┐
│ #   │ VAULT FILE                     │ NAME       │ NAMESPACE  │ SECRETS  │ ACCESSORS  │ ACCESS LIST     │
├─────┼────────────────────────────────┼────────────┼────────────┼──────────┼────────────┼─────────────────┤
│  1  │ internal/k8s/config/samples/sl │ pets       │ (default)  │       11 │          1 │ SLV_EPK_AEAU... │
│     │ v_v1_slv.yaml                  │            │            │          │            │                 │
└─────┴────────────────────────────────┴────────────┴────────────┴──────────┴────────────┴─────────────────┘

📊 Summary: 1 vaults | 11 total secrets | 1 unique accessors
```
<img width="806" height="328" alt="Screenshot 2025-12-10 at 8 53 48 PM" src="https://github.com/user-attachments/assets/3445897b-c625-451d-8b58-b3c847ba7bfa" />

sigsegvd commented:
Hi @realnits ,
Just a small suggestion.
1. Can we have the `slv vault ls -r` functionality as it was before? Listing just the vault files. Because displaying details here too feels redundant.
2. Can we remove the accessors list from the detailed table view? If someone wants to view the list of accessors, they can use the `slv vault -v /path/to/vault.slv.yaml` command. 
Otherwise, this looks perfect. 

Thanks in advance for the changes.

realnits committed:
Simplify vault list command based on feedback

- Restore simple list view (slv vault ls -r) to show only vault files
- Remove Access List column from detailed table view
- Keep only accessor count in ACCESSORS column
- Add documentation tip to use 'slv vault -v' for viewing accessor details
- Maintain cell wrapping for better terminal readability

sigsegvd commented:
This is great @realnits. I can still see some overflow when the font size is large but I don't think there's much we can do about it. Thanks. Will merge the PR now.
